misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestProcessor.java

```diff
@@ -576,6 +574,10 @@ private String getFieldName(SpecificFieldNode specificField) {
      * @return String
      */
     private String getExecutePath(Module module) {
+        if (isSingleFileProject(module.project())) {
+            String fileName = module.project().sourceRoot().getFileName().toString();
```

Comment:
`module.project().sourceRoot().getFileName()` should be wrapped with `Optional.of`

Reply:
This has to be done as follows:
```java
String fileName = Optional.of(module.project().sourceRoot().getFileName()).get().toString();
return FileUtils.getFileNameWithoutExtension(fileName);
```